src/components/tabs/ScatterPlotTab.tsx

```diff
@@ -0,0 +1,709 @@
+  const setRangeType = (newRangeType: RangeType): void => {
+    if (newRangeType === rangeType) {
+      return;
+    }
+    setIsRendering(true);
+    startTransition(() => {
+      _setRangeType(newRangeType);
+    });
+  };
+
+  const [xAxisFeatureName, _setXAxisFeatureName] = useState<string | null>(null);
+  const setXAxisFeatureName = (newFeatureName: string | null): void => {
+    if (newFeatureName === xAxisFeatureName) {
+      return;
+    }
+    setIsRendering(true);
+    startTransition(() => {
+      _setXAxisFeatureName(newFeatureName);
+    });
+  };
+  const [yAxisFeatureName, _setYAxisFeatureName] = useState<string | null>(null);
+  const setYAxisFeatureName = (newFeatureName: string | null): void => {
+    if (newFeatureName === yAxisFeatureName) {
+      return;
+    }
+    setIsRendering(true);
+    startTransition(() => {
+      _setYAxisFeatureName(newFeatureName);
+    });
+  };
+
+  //////////////////////////////////
+  // Click Handlers
+  //////////////////////////////////
+
+  /**
+   * Returns true if a Plotly mouse event took place over a histogram subplot.
+   */
+  function isHistogramEvent(eventData: Plotly.PlotMouseEvent): boolean {
+    return eventData.points.length > 0 && eventData.points[0].data.type === "histogram";
+  }
+
+  // Add click event listeners to the plot. When clicking a point, find the track and jump to
+  // that frame.
+  const onClickPlot = useCallback(
+    (eventData: Plotly.PlotMouseEvent): void => {
+      if (!dataset || eventData.points.length === 0 || isHistogramEvent(eventData)) {
+        return;
+      }
+
+      const point = eventData.points[0];
+      const objectId = Number.parseInt(point.data.ids[point.pointNumber], 10);
+      const trackId = dataset.getTrackId(objectId);
+      const frame = dataset.times ? dataset.times[objectId] : undefined;
+      if (frame !== undefined) {
+        props.setFrame(frame).then(() => {
+          props.findTrack(trackId, false);
+        });
+      } else {
+        // Jump to first frame where the track is valid
+        props.findTrack(trackId, true);
+      }
+    },
+    [dataset, props.findTrack, props.setFrame]
+  );
+
+  useEffect(() => {
+    plotRef.current?.on("plotly_click", onClickPlot);
+    return () => {
+      plotRef.current?.removeAllListeners("plotly_click");
+    };
+  }, [plotRef.current, onClickPlot]);
+
+  //////////////////////////////////
+  // Helper Methods
+  //////////////////////////////////
+
+  /** Retrieve feature data, if it exists. Accounts for the artificially-added time feature. */
+  const getData = useCallback(
+    (featureName: string | null, dataset: Dataset | null): Uint32Array | Float32Array | undefined => {
+      if (featureName === null || dataset === null) {
+        return undefined;
+      }
+      if (featureName === TIME_FEATURE.name) {
+        return dataset.times || undefined;
+      }
+      return dataset.getFeatureData(featureName)?.data;
+    },
+    []
+  );
```

Comment:
I think I've been overusing `useCallback` to be honest 😅 Ended up removing this because you're right, I don't think there's a real benefit to using it here.

Reply:
I've absolutely done this too 🙃 it's just so easy to throw around a function just in case. Relative to most other problems you can create with code, overusing `useCallback` is probably miniscule in terms of performance/correctness, but still worth thinking about